src/org/omegat/filters2/IParseCallback.java

```diff
@@ -55,24 +55,46 @@ public interface IParseCallback {
      * @param filter
      *            filter which produces entry
      * @param protectedParts
-     *            (since 3.0.6) protected parts
+     *            protected parts
      */
     void addEntryWithProperties(String id, String source, String translation, boolean isFuzzy, String[] props,
             String path, IFilter filter, List<ProtectedPart> protectedParts);
 
     /**
-     * Read entry from source file, with single "comment" property. Convenience
-     * method for
+     * Read entry from a source file, with single "comment" property.
+     * Convenience method for
      * {@link #addEntryWithProperties(String, String, String, boolean, String[], String, IFilter, List)}.
      */
     void addEntry(String id, String source, String translation, boolean isFuzzy, String comment, String path,
             IFilter filter, List<ProtectedPart> protectedParts);
 
     /**
-     * Old call without path, for compatibility with OmegaT &lt; 2.5.0
+     * This method is called by filters to add new entry in OmegaT after read it
+     * from a source file.
+     * <p>
+     * Old call without a path, for compatibility. Comment is converted to a
+     * property.
+     *
+     * @param id
+     *            ID of entry, if a format supports it
+     * @param source
+     *            Translatable source string
+     * @param translation
+     *            translation of the source string, if a format supports it
+     * @param isFuzzy
+     *            flag for fuzzy translation. If a translation is fuzzy, it is
+     *            not added to the projects TMX, but it is added to the
+     *            generated 'reference' TMX, a special TMX that is used as extra
+     *            reference during translation.
+     * @param comment
+     *            entry's comment, if format supports it
+     * @param filter
+     *            filter which produces entry
      */
-    void addEntry(String id, String source, String translation, boolean isFuzzy, String comment,
-            IFilter filter);
+    default void addEntry(String id, String source, String translation, boolean isFuzzy, String comment,
```

Comment:
Compatibility is important.  We need to check whether a change from standard method to `default` keeps its backward compatibility.

Reply:
Binary compatibility guide said
https://docs.oracle.com/javase/specs/jls/se17/html/jls-13.html

> 13.5.7. Interface Method Declarations
> (snip)
> Adding a default method, **or changing a method from abstract to default, does not break compatibility with pre-existing binaries**, but may cause an IncompatibleClassChangeError if a pre-existing binary attempts to invoke the method. This error occurs if the qualifying interface of the method invocation, K, is a subinterface of two interfaces, I and J, where both I and J declare a default method with the same signature and result, and neither I nor J is a subinterface of the other.

Reply:
It considered as safe, because the interface `IParseCallback` is for OmegaT core internal, and rarely, or nowhere, used outside the core.

Reply:
Question is resolved.